x/evm/keeper/keeper.go

```diff
@@ -258,11 +264,16 @@ func (k *Keeper) GetNonce(ctx sdk.Context, addr common.Address) uint64 {
 	return acct.GetSequence()
 }
 
-// GetBalance load account's balance of gas token
+// GetBalance load account's balance of gas token.
 func (k *Keeper) GetBalance(ctx sdk.Context, addr common.Address) *big.Int {
 	cosmosAddr := sdk.AccAddress(addr.Bytes())
-	baseDenom := config.GetDenom()
-	coin := k.bankKeeper.GetBalance(ctx, cosmosAddr, baseDenom)
+
+	// The registered EVM coin is the one used as gas token.
+	evmDenom := config.GetEVMCoinDenom()
+
+	// Get the balance via bank wrapper to convert it to 18 decimals if needed.
+	coin := k.bankWrapper.GetBalance(ctx, cosmosAddr, evmDenom)
```

Comment:
here we can use the `GetEVMCoinBalance` func
```suggestion
	// Get the balance via bank wrapper to convert it to 18 decimals if needed.
	coin := k.bankWrapper.GetEVMCoinBalance(ctx, cosmosAddr)
```